lib/PrintAsClang/PrintAsClang.cpp

```diff
@@ -393,9 +395,90 @@ static int compareImportModulesByName(const ImportModuleTy *left,
   return 1;
 }
 
+// Collect the set of header includes needed to import the given Clang module
+// into an ObjectiveC program. Modeled after collectModuleHeaderIncludes in the
+// Clang frontend (FrontendAction.cpp)
+// Augment requiredTextualIncludes with the set of headers required.
+static void collectClangModuleHeaderIncludes(
+    const clang::Module *clangModule, clang::FileManager &fileManager,
+    llvm::SmallSet<llvm::SmallString<32>, 10> &requiredTextualIncludes) {
+
+  auto addHeader = [&](const clang::Module *module, StringRef header) {
+    llvm::SmallString<32> fullHeaderName = header;
+    if (clangModule->IsFramework) {
+      // Frameworks store their headers in an explicit header directory
+      // Transform "Headers/Foundation.h" -> "Foundation/Foundation.h"
+      llvm::sys::path::replace_path_prefix(fullHeaderName, "Headers",
+                                           clangModule->Name);
+    } else {
```

Comment:
These prefixes are a little fragile, I prepend the directory name under the assumption that more often than not, headers are found somewhere like `.../MacOSX.sdk/usr/include/objc/NSObject.h`, and included `#import <objc/NSObject.h>` but if the header lies directly on the configured include path and imported directly (`#import <NSObject.h>`), it breaks down (`#import <include/NSObject.h>`) is imported instead. Not sure how to get around this without access to informations about the system consuming the header.

Reply:
I'm thinking maybe explicitly forbidding the "include" prefix would handle the majority of cases.